packages/flame_riverpod/lib/src/consumer.dart

```diff
@@ -139,8 +139,30 @@ mixin RiverpodGameMixin<W extends World> on FlameGame<W> {
   /// Used to facilitate [Component] access to the [ProviderContainer].
   GlobalKey<RiverpodAwareGameWidgetState>? widgetKey;
 
+  final ComponentRef ref = ComponentRef(game: null);
   final List<void Function()> _onBuildCallbacks = [];
 
+  /// Adds a callback method to be invoked in the build method of
+  /// [RiverpodAwareGameWidgetState].
+  void addToGameWidgetBuild(Function() cb) {
```

Comment:
Hmm, not too fond of this one, it feels like this should be possible to handle under the hood for the user?
What do you think @markvideon?

Reply:
@spydon there doesn't appear to be any mechanism for removing callbacks from _onBuildCallbacks in RiverpodGameMixin at the moment

Reply:
The problem here is that some ComponentRef methods, like read or watch, return a value. So we need to wrap these calls (like read or watch) in a callback, as they are by design executed later whenever the game builds. So something like
```dart
void onMount() {
    final myValue = ref.watch(myProvider);
}
```
will not work as the actual watching of myProvider happens later when the game builds. As to the exact reason why we do this behaviour - i don't really know. Maybe @markvideon can explain it better.

Reply:
> The problem here is that some ComponentRef methods, like read or watch, return a value. So we need to wrap these calls (like read or watch) in a callback, as they are by design executed later whenever the game builds. So something like
> 
> ```dart
> void onMount() {
>     final myValue = ref.watch(myProvider);
> }
> ```
> 
> will not work as the actual watching of myProvider happens later when the game builds. As to the exact reason why we do this behaviour - i don't really know. Maybe @markvideon can explain it better.

If you take a look at the implementation of watch, doing this _will_ work in the sense that it will create a subscription and obtain a value automatically, but it _won't_ work in the sense that the subscription can not be expected to persist in the way that a user might expect if they're coming from flutter_riverpod. In flutter_riverpod, all current subscriptions are terminated before an application's ConsumerWidget/ConsumerStatefulWidget build code is executed - which, if it contains calls to watch or listen operations, re-creates the subscriptions.